### PR TITLE
Use the bag tracker in the bags API

### PR DESCRIPTION
**Pull request by alexwlchan**
Another step towards https://github.com/wellcomecollection/platform/issues/4534

### CODE REVIEW

File: bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/BagsApi.scala [hunk 1]
```diff
@@ -51,59 +88,16 @@ trait BagsApi extends LargeResponses with LookupBag with LookupBagVersions {
         )
 
         get {
-          parameter('version.as[String] ?) {
-            case None => getLatestBag(bagId = bagId)
-            case Some(versionString) =>
-              getBag(bagId = bagId, versionString = versionString)
+          parameter('version.as[String] ?) { maybeVersionString =>
+            withFuture {
+              lookupBagBetter(bagId = bagId, maybeVersionString = maybeVersionString)
```
Comment (kenoir) [inline bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/BagsApi.scala:93]: Can we rename this now the previous function is replaced?

File: bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/models/DisplayBagVersionList.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package uk.ac.wellcome.platform.storage.bags.api.models
+
+import java.net.URL
+
+import io.circe.generic.extras.JsonKey
+import uk.ac.wellcome.platform.archive.bag_tracker.models.{BagVersionEntry, BagVersionList}
+import uk.ac.wellcome.platform.archive.common.bagit.models.BagId
+import uk.ac.wellcome.platform.archive.common.storage.models.StorageManifest
+
+case class DisplayBagVersionEntry(
```
Comment (kenoir) [inline bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/models/DisplayBagVersionList.scala:10]: Where are these used?

File: bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/BagsApi.scala (hunk 1 above)
Comment (alexwlchan) [reply bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/BagsApi.scala]: Oops! Will fix.

File: bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/models/DisplayBagVersionList.scala (hunk 2 above)
Comment (alexwlchan) [reply bags_api/src/main/scala/uk/ac/wellcome/platform/storage/bags/api/models/DisplayBagVersionList.scala:13]: They're used to display a list of version, e.g. if you query

```
GET /bags/digitised/b1234/versions
```

They're renamed to match the models from the internal bag tracker classes.

**Pull request merged**